x-pack/plugins/aiops/public/embeddables/pattern_analysis/pattern_analysis_initializer.tsx

```diff
@@ -0,0 +1,402 @@
+              onValidationChange={setIsFormValid}
+            />
+          </DataSourceContextProvider>
+        </EuiForm>
+      </EuiFlyoutBody>
+
+      <EuiFlyoutFooter>
+        <EuiFlexGroup justifyContent="spaceBetween">
+          <EuiFlexItem grow={false}>
+            <EuiButtonEmpty
+              color="primary"
+              size="m"
+              onClick={onCancel}
+              data-test-subj="aiopsPatternAnalysisCancelButton"
+            >
+              <FormattedMessage
+                id="xpack.aiops.embeddablePatternAnalysis.config.cancelButtonLabel"
+                defaultMessage="Cancel"
+              />
+            </EuiButtonEmpty>
+          </EuiFlexItem>
+          <EuiFlexItem grow={false}>
+            <EuiButton
+              onClick={onCreate.bind(null, updatedProps)}
+              fill
+              aria-label={i18n.translate(
+                'xpack.aiops.embeddablePatternAnalysis.config.applyFlyoutAriaLabel',
+                {
+                  defaultMessage: 'Apply changes',
+                }
+              )}
+              isDisabled={!isFormValid}
+              iconType="check"
+              data-test-subj="aiopsPatternAnalysisConfirmButton"
+            >
+              <FormattedMessage
+                id="xpack.aiops.embeddablePatternAnalysis.config.applyAndCloseLabel"
+                defaultMessage="Apply and close"
+              />
+            </EuiButton>
+          </EuiFlexItem>
+        </EuiFlexGroup>
+      </EuiFlyoutFooter>
+    </>
+  );
+};
+
+export const FormControls: FC<{
+  formInput: PatternAnalysisEmbeddableRuntimeState;
+  onChange: (update: PatternAnalysisEmbeddableRuntimeState) => void;
+  onValidationChange: (isValid: boolean) => void;
+}> = ({ formInput, onChange, onValidationChange }) => {
+  const dataViewId = formInput.dataViewId;
+  const {
+    data: { dataViews },
+  } = useAiopsAppContext();
+  const [fields, setFields] = useState<DataViewField[]>([]);
+  const [selectedField, setSelectedField] = useState<DataViewField | null>(null);
+  const [isDataViewTimeBased, setIsDataViewTimeBased] = useState(true);
+
+  const randomSampler = useMemo(() => {
+    return new RandomSampler({
+      randomSamplerMode: formInput.randomSamplerMode ?? RANDOM_SAMPLER_OPTION.ON_AUTOMATIC,
+      setRandomSamplerMode: () => {},
+      randomSamplerProbability: formInput.randomSamplerProbability ?? DEFAULT_PROBABILITY,
+      setRandomSamplerProbability: () => {},
+    });
+  }, [formInput.randomSamplerMode, formInput.randomSamplerProbability]);
+  const randomSamplerMode = useObservable(randomSampler.getMode$(), randomSampler.getMode());
+  const randomSamplerProbability = useObservable(
+    randomSampler.getProbability$(),
+    randomSampler.getProbability()
+  );
+
+  useEffect(
+    function initFields() {
+      dataViews.get(dataViewId).then((dataView) => {
+        const isTimeBased = dataView.isTimeBased();
+        setIsDataViewTimeBased(isTimeBased);
```

Comment:
can potentially set state for unmounted component

Reply:
Updated in https://github.com/elastic/kibana/pull/186539/commits/a3008dc3d0f23cb78be2899bcb59236eec21d846